clients/python-wrapper/lakefs/object_io.py

```diff
@@ -133,29 +136,65 @@ def _extract_etag_from_response(headers) -> str:
         etag = headers.get("ETag", "").strip(' "')
         return etag
 
-    def _upload(self, content, pre_sign, content_type: Optional[str] = None,
-                metadata: Optional[dict[str, str]] = None) -> ObjectStats:
-        if not pre_sign:
-            raise UnsupportedOperationException("Upload currently supported only in pre-sign mode")
-
-        headers = {}
-        if content_type is not None:
-            headers["Content-Type"] = content_type
+    def _upload_raw(self, content, content_type, metadata):
```

Comment:
typing hints would help understand what content is acceptable (does this support a streaming upload?)

Reply:
Currently not been tested - I have another issue for this to ensure we support streams:
https://github.com/treeverse/lakeFS/issues/6964

Reply:
The public method `create` provides the type hints for the parameters

Reply:
Done - though we will probably need to modify it once we support streams